src/components/screens/SwapScreen/hooks/useSwapTransaction.ts

```diff
@@ -0,0 +1,146 @@
+import { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import { getTokenFromBalance } from "components/screens/SwapScreen/helpers";
+import { NETWORKS } from "config/constants";
+import { logger } from "config/logger";
+import {
+  SWAP_ROUTES,
+  SwapStackParamList,
+  ROOT_NAVIGATOR_ROUTES,
+  MAIN_TAB_ROUTES,
+} from "config/routes";
+import { PricedBalance, NativeToken, AssetToken } from "config/types";
+import { ActiveAccount } from "ducks/auth";
+import { SwapPathResult } from "ducks/swap";
+import { useTransactionBuilderStore } from "ducks/transactionBuilder";
+import { useState } from "react";
+
+interface SwapTransactionParams {
+  swapAmount: string;
+  swapFromTokenBalance: PricedBalance | undefined;
+  swapToTokenBalance: PricedBalance | undefined;
+  pathResult: SwapPathResult | null;
+  account: ActiveAccount | null;
+  swapFee: string;
+  swapTimeout: number;
+  network: NETWORKS;
+  navigation: NativeStackNavigationProp<
+    SwapStackParamList,
+    typeof SWAP_ROUTES.SWAP_AMOUNT_SCREEN
+  >;
+  resetSwap: () => void;
+}
+
+interface UseSwapTransactionResult {
+  isProcessing: boolean;
+  executeSwap: () => Promise<void>;
+  prepareSwapTransaction: () => Promise<void>;
```

Comment:
should we rename it to something else? "prepare" makes me think it's handling a soroban transaction which is not the case here